File: Leetcode/JS/Leetcode299-猜数字游戏.js
/**
 * @param {string} secret
 * @param {string} guess
 * @return {string}
 */
 var getHint = function(secret, guess) {
    let st = new Map();
    let bull = 0,cow = 0;
    for(let i = 0;i < secret.length;i++){
        if(guess[i] == secret[i]) bull++;
        else {
            if(st.has(secret[i])) st.set(secret[i],st.get(secret[i]) + 1);
            else st.set(secret[i],1);
        }  
    }
    for(let i = 0;i < guess.length;i++){
        if(guess[i] != secret[i] && st.get(guess[i])){
            cow++;
            st.set(guess[i],st.get(guess[i]) - 1);
        }
    }
    return String(bull) + "A" + String(cow) + "B";
};

File: Leetcode/Rust/Leetcode299-猜数字游戏.rs
impl Solution {
    pub fn get_hint(secret: String, guess: String) -> String {
        let mut st = vec![0;26];
        let (mut bull,mut cow) = (0,0);
        let (secret,guess) = (secret.as_bytes(),guess.as_bytes());
        for i in 0..secret.len(){
            if secret[i] == guess[i]{
                bull += 1;
            }else{
                st[(secret[i] - b'0') as usize] += 1;
            }
        }
        for i in 0..guess.len(){
            if guess[i] != secret[i] && st[(guess[i] - b'0') as usize] > 0{
                st[(guess[i] - b'0') as usize] -= 1;
                cow += 1;
            }
        }
        bull.to_string() + "A" + &cow.to_string() + "B"
    }
}